### PR TITLE
cronet: allow application to provide all threads

### PR DESCRIPTION
**Pull request by dapengzhang0**
This is the cronet counterpart of #4139 and #4150.

### CODE REVIEW

File: cronet/src/main/java/io/grpc/cronet/CronetChannelBuilder.java [hunk 1]
```diff
@@ -161,12 +165,29 @@ public final CronetChannelBuilder setTrafficStatsUid(int uid) {
...
+    return this;
+  }
+
   @Override
   protected final ClientTransportFactory buildTransportFactory() {
     return new CronetTransportFactory(
         new TaggingStreamFactory(
             cronetEngine, trafficStatsTagSet, trafficStatsTag, trafficStatsUidSet, trafficStatsUid),
-        MoreExecutors.directExecutor(),
+        MoreExecutors.directExecutor(), scheduledExecutorService,
```
Comment (ericgribkoff) [inline cronet/src/main/java/io/grpc/cronet/CronetChannelBuilder.java]: This should be two separate lines

File: cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java [hunk 2]
```diff
@@ -16,15 +16,22 @@
...
+import static org.junit.Assert.assertSame;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
 
 import io.grpc.CallOptions;
 import io.grpc.Metadata;
 import io.grpc.MethodDescriptor;
 import io.grpc.cronet.CronetChannelBuilder.CronetTransportFactory;
+import io.grpc.internal.ClientTransportFactory;
+import io.grpc.internal.ConnectionClientTransport;
```
Comment (ericgribkoff) [inline cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java]: Is this used?

File: cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java [hunk 3]
```diff
@@ -73,4 +80,32 @@ public void alwaysUsePut_defaultsToFalse() throws Exception {
...
+  @Test
+  public void scheduledExecutorService_custom() {
+    CronetChannelBuilder builder = CronetChannelBuilder.forAddress("address", 1234, mockEngine);
+    ScheduledExecutorService scheduledExecutorService = mock(ScheduledExecutorService.class);
+
+    CronetChannelBuilder builder1 = builder.scheduledExecutorService(scheduledExecutorService);
+    assertSame(builder, builder1);
+
+    ClientTransportFactory clientTransportFactory = builder1.buildTransportFactory();
+
```
Comment (ericgribkoff) [inline cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java]: nit: unnecessary empty line

File: cronet/src/main/java/io/grpc/cronet/CronetChannelBuilder.java (hunk 1 above)
Comment (dapengzhang0) [reply cronet/src/main/java/io/grpc/cronet/CronetChannelBuilder.java]: Done.

File: cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java (hunk 2 above)
Comment (dapengzhang0) [reply cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java]: Removed.

File: cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java (hunk 3 above)
Comment (dapengzhang0) [reply cronet/src/test/java/io/grpc/cronet/CronetChannelBuilderTest.java]: Done.

**Pull request merged**